components/director/internal/systemfetcher/service.go

```diff
@@ -227,21 +239,75 @@ func (s *SystemFetcher) processSystemsForTenant(ctx context.Context, tenantMappi
 	return nil
 }
 
-func (s *SystemFetcher) convertSystemToAppRegisterInput(sc System) (model.ApplicationRegisterInput, string) {
-	initStatusCond := model.ApplicationStatusConditionInitial
-	baseURL := sc.BaseURL
+func (s *SystemFetcher) convertSystemToAppRegisterInput(ctx context.Context, sc System) (*model.ApplicationRegisterInputWithTemplate, error) {
+	if len(sc.TemplateID) > 0 {
+		return s.appRegisterInputFromTemplate(ctx, sc)
+	}
 
 	appRegisterInput := model.ApplicationRegisterInput{
-		Name:            sc.DisplayName,
-		Description:     &sc.ProductDescription,
-		BaseURL:         &baseURL,
-		ProviderName:    &sc.InfrastructureProvider,
-		StatusCondition: &initStatusCond,
-		SystemNumber:    &sc.SystemNumber,
-		Labels: map[string]interface{}{
-			"managed": "true",
-		},
-	}
-
-	return appRegisterInput, sc.TemplateID
+		Name: sc.DisplayName,
+	}
+
+	return enrichAppRegisterInput(appRegisterInput, sc), nil
+}
+
+func (s *SystemFetcher) appRegisterInputFromTemplate(ctx context.Context, sc System) (*model.ApplicationRegisterInputWithTemplate, error) {
+	appTemplate, err := s.appTemplateService.Get(ctx, sc.TemplateID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while getting application template with ID %s", sc.TemplateID)
+	}
+
+	inputValues := getTemplateInputs(appTemplate, sc)
+	appRegisterInputJSON, err := s.appTemplateService.PrepareApplicationCreateInputJSON(appTemplate, inputValues)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while preparing ApplicationRegisterInput JSON from Application Template with name %s", appTemplate.Name)
+	}
+
+	appRegisterInput, err := s.appConverter.CreateInputJSONToModel(ctx, appRegisterInputJSON)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while preparing ApplicationRegisterInput model from Application Template with name %s", appTemplate.Name)
+	}
+
+	return enrichAppRegisterInput(appRegisterInput, sc), nil
+}
+
+func enrichAppRegisterInput(input model.ApplicationRegisterInput, sc System) *model.ApplicationRegisterInputWithTemplate {
+	initStatusCond := model.ApplicationStatusConditionInitial
+
+	input.Name = sc.DisplayName
```

Comment:
Do we need `input.Name = sc.DisplayName` since the name will be already populated in `s.appConverter.CreateInputJSONToModel` with the `name` placeholder value which is `s.DisplayName`.

And also, in the `input.Description = &sc.ProductDescription` assignment, the description will be populated in `s.appConverter.CreateInputJSONToModel` with the `display-name` placeholder value and if this value is different from `&sc.ProductDescription` it will be overridden. Is this the expected behavior?

Reply:
I assumed we want the data from the external registry, since the templates we currently have provide almost no real data, just system names, and types. We can sync this with the team.